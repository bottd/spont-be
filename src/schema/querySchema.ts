import * as graphql from 'graphql';
import { UserType, LocationType } from './types';
import {
  selectUserByID,
  selectLocationByID,
  selectAllLocations,
} from './utils';

const { GraphQLObjectType, GraphQLString, GraphQLNonNull, GraphQLList } = graphql;

export const querySchema = new GraphQLObjectType({
  name: 'RootQuery',
  fields: {
    user: {
      type: UserType,
      args: { id: { type: GraphQLNonNull(GraphQLString) } },
      resolve(parentValue, args) {
        return selectUserByID(args.id);
      },
    },
    location: {
      type: LocationType,
      args: { id: { type: GraphQLNonNull(GraphQLString) } },
      resolve(parentValue, args) {
        return selectLocationByID(args.id);
      },
    },
    locations: {
      type: new GraphQLList(LocationType),
      resolve(parentValue, args) {
        return selectAllLocations();
      },
    },
  },
});
