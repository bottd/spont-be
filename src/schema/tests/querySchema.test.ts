import { querySchema } from '../querySchema';

describe('querySchema', () => {
  it('should match snapshot', () => {
    expect(querySchema).toMatchSnapshot();
  });
});
