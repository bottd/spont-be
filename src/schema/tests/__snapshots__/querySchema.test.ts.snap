// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`querySchema should match snapshot 1`] = `"RootQuery"`;
